src/libNode/MicroBlockPreProcessing.cpp

```diff
@@ -68,9 +68,20 @@ bool Node::ComposeMicroBlock() {
   uint32_t version = BLOCKVERSION::VERSION1;
   uint32_t shardId = m_myshardId;
   uint256_t gasLimit = MICROBLOCK_GAS_LIMIT;
-  uint256_t gasUsed = 1;
+  uint256_t gasUsed = m_gasUsedTotal;
+  uint256_t rewards = 0;
+  if (m_mediator.GetIsVacuousEpoch() &&
+      m_mediator.m_ds->m_mode != DirectoryService::IDLE) {
+    if (!SafeMath<uint256_t>::add(m_mediator.m_ds->m_totalTxnFees,
```

Comment:
Should you return ?

Reply:
if so just remain the rewards 0. safe enough though

Reply:
ok